### PR TITLE
fix Cart.getShippingTotalByShop and use it in Cart.getSubTotal and in…

### PR DESCRIPTION
**Pull request by prosf**
fix issue #3190

### CODE REVIEW

**Comment by spencern** [comment]
@zenweasel I think you might be better suited to review this PR as it's dealing directly with your cart/order transforms

File: lib/collections/transform/cartOrder.js [hunk 1]
```diff
@@ -82,11 +87,24 @@ export const cartOrderTransform = {
...
-    }
-    return billingObject;
+    // The previous implementation  can only be used if an invoice object already exists( while Invoice
+    // to be generated, needs this functionality of getShippingTotalByShop in the first place)
+    // So a similar approach to getSubtotalByShop, like the current one, seems okay for now.
+    // const billingObject = {};
+    // for (const billingItem of this.billing) {
+    //   billingObject[billingItem.shopId] = billingItem.invoice.shipping;
+    // }
+    // return billingObject;
```
Comment (zenweasel) [inline lib/collections/transform/cartOrder.js]: Shouldn't leave commented code. Let's remove?

File: lib/collections/transform/cartOrder.js [hunk 2]
```diff
@@ -82,11 +87,24 @@ export const cartOrderTransform = {
    * @returns {{Object}} - Total price of shipping, broken down by shop
    */
   getShippingTotalByShop() {
-    const billingObject = {};
-    for (const billingItem of this.billing) {
-      billingObject[billingItem.shopId] = billingItem.invoice.shipping;
-    }
-    return billingObject;
+    // The previous implementation  can only be used if an invoice object already exists( while Invoice
```
[the remark this answers is not in the document]
Comment (prosf) [reply lib/collections/transform/cartOrder.js]: Indeed, the right place for this comment is in the pr and not in the code.
Nevertheless what I mean here is that previously, getShippingTotalByShop couldn't be called before the buyer charging (e.g couldn't be used in order-preview of checkout) as it required access to the billing.$.invoice which gets created and is accessible only after the buyer-charging.

**Comment by prosf** [comment]
@zenweasel  Style changes done

**Pull request merged**